### PR TITLE
[feat]: validate github action added

### PR DESCRIPTION
**Pull request by tarun-khanna**
# Description

A new "validate" github action added for pull requests. 
This github action will run the below tasks 
- **run eslint:** to ensure no code with eslint errors is merged to master
- **run tsc:** to ensure code with proper types is merged to master
- **run build:** to ensure we are able to create build successfully  

| Title                  | --- |
| ---------------------- | --- |
| JIRA link              | https://razorpay.atlassian.net/browse/ATLAS-34 |
| Slack thread (if any)  | NA  |
| Product spec           | NA  |
| Tech spec/One-pager    | NA  |
| Bundle Size Difference | NA  |

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Is your change backward compatible ?
- [ ] Tested on major browsers (Chrome, Firefox, Safari, IE) ?
- [ ] Tested in a consumer application(s) ?

## Additional Notes

Any additional information that would be helpful for the reviewer.

## Checklist:

- [x] Add Jira ID(s) in PR title and in the description?
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code is written in TypeScript and follows the coding standards of the project.
- [ ] I have added relevant documentation and updated the README if necessary.
- [x] My commit messages are clear and follow the project's commit message conventions.
- [ ] Is any external library added?
- [x] My changes do not introduce any new runtime errors or warnings.
- [x] All relevant unit tests have been added/updated in my PR for the change.
- [ ] Any Screenshots (mobile & desktop) required for PR? If yes, have you added the respective screenshots ?
- [ ] Any manual dev testing done by you on beta/func environment?
- [ ] Reviewer added (SLA 2 days)

## Reviewer Checklist

- [ ] Sufficient QA/Dev-Testing is done with proof (test cases list)
- [ ] Ensure that the change in bundle size falls within the acceptable range.

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        run: yarn tsc
+  build:
+    name: Run Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 20
+      - name: Install dependencies
```
Comment (avinashdvv) [inline .github/workflows/validate.yml:37]: change: i don't see much of use of the multiple `Install dependencies`? can you please elaborate why this is needed

**Review by avinashdvv — CHANGES_REQUESTED** [review]
Mostly looks good, please address the comments

File: .github/workflows/validate.yml (hunk 1 above)
Comment (tarun-khanna) [reply .github/workflows/validate.yml:37]: dependencies are required for each job to run successfully.

**Review by RgnDunes — APPROVED** [review]
Looks good to me

**Pull request merged**